ostd/src/arch/x86/boot/smp.rs

```diff
@@ -186,17 +191,27 @@ unsafe fn fill_boot_info_ptr(info_ptr: *const PerApRawInfo) {
 
 /// # Safety
 ///
-/// The caller must ensure the pointer to be filled is valid to write.
+/// 1. This function should only be called once on the BSP.
+/// 2. This function should be called before any AP is booted.
+/// 3. This function should be called after the kernel page table is activated
+///    on the BSP.
+/// 4. The `pt_ptr` must be the physical address of a valid boot page table's
+///    root.
```

Comment:
`fill_boot_pt_ptr` does not require the physical address to be a valid boot page table. **You can call it with any address, as long as you don't call `send_boot_ipis`.**

In order words, this condition is a safety precondition of `send_boot_ipis`, not `fill_boot_pt_ptr`.
https://github.com/asterinas/asterinas/blob/45b5bd39f1aa61bfc6f00c509913b7ec10a3a68f/ostd/src/arch/x86/boot/smp.rs#L255-L257